exporter/datadog/datadog_test.go

```diff
@@ -0,0 +1,413 @@
+		}
+	}
+}
+
+func TestExporter(t *testing.T) {
+	var (
+		tagKey, _ = tag.NewKey("key")
+		tagValue  = "value"
+		m, _      = stats.Int64("m", "description connections", "")
+		want      = int64(1)
+	)
+
+	testCases := map[string]struct {
+		Name        string
+		Want        int64
+		Aggregation view.Aggregation
+		Validate    func(t *testing.T, m *Mock, name string, want int64)
+	}{
+		"count": {
+			Name:        "counts",
+			Want:        1,
+			Aggregation: view.CountAggregation{},
+			Validate:    validateCount,
+		},
+		"sum": {
+			Name:        "sum",
+			Want:        1,
+			Aggregation: view.SumAggregation{},
+			Validate:    validateCount,
+		},
+		"mean": {
+			Name:        "mean",
+			Want:        1,
+			Aggregation: view.MeanAggregation{},
+			Validate:    validateGauge,
+		},
+		"distribution": {
+			Name:        "distribution",
+			Want:        100,
+			Aggregation: view.DistributionAggregation{100},
+			Validate:    validateHistogram,
+		},
+	}
+
+	for label, tc := range testCases {
+		t.Run(label, func(t *testing.T) {
+			var (
+				myView, _   = view.New(tc.Name, "blah", []tag.Key{tagKey}, m, tc.Aggregation)
+				mock        = &Mock{}
+				exporter, _ = NewExporter(WithClient(mock))
+				interval    = 100 * time.Millisecond
+			)
+
+			view.RegisterExporter(exporter)
+			view.SetReportingPeriod(interval)
+
+			if err := myView.Subscribe(); err != nil {
+				t.Fatalf("want nil; got %v", err)
+			}
+
+			// When
+			ctx, _ := tag.New(context.Background(), tag.Insert(tagKey, tagValue))
+
+			stats.Record(ctx, m.M(want))
+			time.Sleep(2 * interval)
+
+			// Then
+			myView.Unsubscribe()
+
+			tc.Validate(t, mock, tc.Name, tc.Want)
+		})
+	}
+}
+
+func makeView(key tag.Key, m stats.Measure, name string, aggregation view.Aggregation) *view.View {
+	v, err := view.New(
+		name,
+		"blah",
+		[]tag.Key{key},
+		m,
+		aggregation,
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	return v
+}
+
+func TestLiveExporter(t *testing.T) {
+	addr := os.Getenv("DATADOG_ADDR")
+	if addr == "" {
+		t.SkipNow()
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	exporter, err := NewExporter(WithStatsAddr(addr))
+	if err != nil {
+		t.Fatalf("want nil; got %v", err)
+	}
+
+	view.RegisterExporter(exporter)
+	view.SetReportingPeriod(3 * time.Second)
+
+	m, err := stats.Int64("my.org/measure/openconns", "open connections", "")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var (
+		key, _       = tag.NewKey("key")
+		sum          = makeView(key, m, "opencensus.sample.sum", view.SumAggregation{})
+		count        = makeView(key, m, "opencensus.sample.count", view.CountAggregation{})
+		mean         = makeView(key, m, "opencensus.sample.mean", view.MeanAggregation{})
+		distribution = makeView(key, m, "opencensus.sample.distribution", view.DistributionAggregation{1, 2, 3, 4, 5})
+	)
+
+	sum.Subscribe()
+	defer sum.Unsubscribe()
+
+	count.Subscribe()
+	defer count.Unsubscribe()
```

Comment:
in general I would not worry too much about unsubscribing from views in tests. as long as you ensure that your view names and measure names are unique to the test, it should not be necessary and doesn't really add anything. we don't expect that in practice people will use Unsubscribe much.